### PR TITLE
Fix segfault when running snap on Ubuntu 18.04 by moving to core18

### PR DESCRIPTION
**Pull request by TravisEz13**

### CODE REVIEW

File: preview/snap/snapcraft.yaml [hunk 1]
```diff
@@ -2,18 +2,8 @@ name: powershell-preview
```
Comment (TravisEz13) [inline preview/snap/snapcraft.yaml:2]: This is the fix to resolve the seg fault.  Other changes are to support building these changes

File: preview/launcher/powershell-wrapper [hunk 2]
```diff
@@ -3,6 +3,6 @@
 # Licensed under the MIT License.
 
 # Create $XDG_RUNTIME_DIR if it doesn't exist
-[ -n "$XDG_RUNTIME_DIR" ] && mkdir -p $XDG_RUNTIME_DIR -m 700
+[ -n "$XDG_RUNTIME_DIR" ] && mkdir -p $XDG_RUNTIME_DIR -m 700 || true
```
Comment (rjmholt) [inline preview/launcher/powershell-wrapper:7]: Why is this needed?

File: preview/snap/snapcraft.yaml (hunk 1 above)
Comment (rjmholt) [reply preview/snap/snapcraft.yaml:2]: Would you be able to give some more information about what was causing the segfault previously and how/where these changes prevent it now?

Comment (SteveL-MSFT) [reply preview/snap/snapcraft.yaml:2]: Can you explain what the fix is?  Is it including the additional libraries?

File: preview/snap/snapcraft.yaml [hunk 3]
```diff
@@ -84,10 +85,25 @@ parts:
       done >> $thirdPartyNoticeFile
       echo "Done building third party notices file - $thirdPartyNoticeFile"
     stage-packages:
-      - libicu55
+      - libicu60
```
Comment (rjmholt) [inline preview/snap/snapcraft.yaml:88]: I assume this is the real fix here, compiling for the newest libicu version to avoid .NET Core picking the wrong one up from elsewhere?

File: tools/releaseBuild/Images/microsoft_powershell_ubuntu18.04/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  apt-get dist-upgrade --yes && \
+  apt-get install --yes \
+    curl \
+    sudo \
+    jq \
+    squashfs-tools \
+    git && \
+  curl -L $(curl -H 'X-Ubuntu-Series: 16' 'https://api.snapcraft.io/api/v1/snaps/details/core' | jq '.download_url' -r) --output core.snap && \
+  mkdir -p /snap/core && unsquashfs -d /snap/core/current core.snap && rm core.snap && \
+#  curl -L $(curl -H 'X-Ubuntu-Series: 16' 'https://api.snapcraft.io/api/v1/snaps/details/core18' | jq '.download_url' -r) --output core18.snap && \
```
Comment (rjmholt) [inline tools/releaseBuild/Images/microsoft_powershell_ubuntu18.04/Dockerfile:20]: Should this be present here as a comment?

File: preview/launcher/powershell-wrapper (hunk 2 above)
Comment (TravisEz13) [reply preview/launcher/powershell-wrapper:7]: some environments (Azure devops) have incorrect permissions on this folder and it cannot be created

File: preview/snap/snapcraft.yaml (hunk 3 above)
Comment (TravisEz13) [reply preview/snap/snapcraft.yaml:88]: The problem was that .NET Core picks individual DLLs' newest to oldest, and it can see into the OS.  so on 18.04 it ends up picking a mix of old and new which don't work together.

File: tools/releaseBuild/Images/microsoft_powershell_ubuntu18.04/Dockerfile (hunk 4 above)
Comment (TravisEz13) [reply tools/releaseBuild/Images/microsoft_powershell_ubuntu18.04/Dockerfile:20]: This is dead code... I'll remove

File: preview/launcher/powershell-wrapper (hunk 2 above)
Comment (rjmholt) [reply preview/launcher/powershell-wrapper:7]: We should maybe add a comment about that

**Review by rjmholt — APPROVED** [review]

File: preview/launcher/powershell-wrapper (hunk 2 above)
Comment (TravisEz13) [reply preview/launcher/powershell-wrapper:7]:
```suggestion
# || true causes bash to ignore errors
[ -n "$XDG_RUNTIME_DIR" ] && mkdir -p $XDG_RUNTIME_DIR -m 700 || true
```

File: preview/snap/snapcraft.yaml (hunk 1 above)
Comment (rjmholt) [reply preview/snap/snapcraft.yaml:2]: To document:

- We were originally building snaps with older dependencies, but dotnet inside the snap would resolve the newest it could find on the machine, breaking snaps on systems with newer inbox versions of things like libicu
- .NET don't see this as their problem since (1) their main scenario is being the only app on the box and (2) even if they're not, they're probably expecting to be a non-classic snap
- To fix this, we now build with newer (newest?) versions of those dependencies, which is something that the snap model is essentially intended for
- However, to make this fix happen, @TravisEz13 had to do work to build the images to assemble the snap with the new dependencies, so it was more work than a simple config addition

I do think we should take this back to .NET -- it strikes me as a fairly serious issue if we fall under their support aegis, since it breaks the only reason to have a snap

**Comment by TravisEz13** [comment]
started https://dev.azure.com/powershell/PowerShell-Snap/_build/results?buildId=36001
